neurodamus/core/_utils.py

```diff
@@ -67,8 +69,36 @@ def rank0_wrapper(*args, **kw):
         return rank0_wrapper
 
 
+class SimulationProgress:
+    def __init__(self):
+        """
+        Class which will set up a timer to perioducally check the amount of time lapsed
+        in the simulation compared to the final tstop value. This is converted into a percentage
+        of the job complete which is then printed to the console.
+        """
+        self.last_time_check = time.time()
+        self.sim_start = self.last_time_check
```

Comment:
In the hoc version, these two vars are initialized for rank 0. Here we do for all ranks. Are they the same ?